### PR TITLE
Adding logging and scoring

### PR DESCRIPTION
**Pull request by arjundc-db**
Signed-off-by: Arjun DCunha <arjun.dcunha@databricks.com>

## What changes are proposed in this pull request?

Adding logging and scoring to the pytorch example.

## How is this patch tested?

Manually using mlflow run .

### Is this a user-facing change?

- [ x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/pytorch/mnist_tensorboard_artifact.py [hunk 1]
```diff
@@ -218,3 +220,23 @@ def log_scalar(name, value, step):
...
+    mlflow.pytorch.log_model(model, artifact_path="pytorch-model", pickle_module=pickle)
+    print(
+        "\nThe model is logged at:\n%s" % os.path.join(mlflow.get_artifact_uri(), "pytorch-model")
+    )
+
+    # Since the model was logged as an artifact, it can be loaded to make predictions
+    loaded_model = mlflow.pytorch.load_model(mlflow.get_artifact_uri("pytorch-model"))
+
+    # Extract a few examples from the test dataset to evaulate on
+    _, (eval_data, eval_labels) = next(enumerate(test_loader))
```
Comment (harupy) [inline examples/pytorch/mnist_tensorboard_artifact.py:235]: Can we remove `enumerate` here?

File: examples/pytorch/mnist_tensorboard_artifact.py [hunk 2]
```diff
@@ -218,3 +220,23 @@ def log_scalar(name, value, step):
         "\nLaunch TensorBoard with:\n\ntensorboard --logdir=%s"
         % os.path.join(mlflow.get_artifact_uri(), "events")
     )
+
+    # Log the model as an artifact of the MLflow run.
+    print("\nLogging the trained model as a run artifact...")
+    mlflow.pytorch.log_model(model, artifact_path="pytorch-model", pickle_module=pickle)
```
Comment (harupy) [inline examples/pytorch/mnist_tensorboard_artifact.py:226]: `pickle_module=pickle` is to suppress the error reported in for https://github.com/mlflow/mlflow/issues/3199?

Comment (arjundc-db) [reply examples/pytorch/mnist_tensorboard_artifact.py:226]: Yes, I faced the exact same issue. Using the work-around you suggested on #3199. Thank you for it!

File: examples/pytorch/mnist_tensorboard_artifact.py (hunk 1 above)
Comment (arjundc-db) [reply examples/pytorch/mnist_tensorboard_artifact.py:235]: I used enumerate to get a single batch from the test loader. Shall I replace it with `iter`? Is there a reasoning that I am missing?

Comment (harupy) [reply examples/pytorch/mnist_tensorboard_artifact.py:235]: `iter` looks better to me because we can remove `_`.

**Review by smurching — COMMENTED** [review]
Looks good pending @harupy's comment!

File: examples/pytorch/mnist_tensorboard_artifact.py (hunk 1 above)
Comment (arjundc-db) [reply examples/pytorch/mnist_tensorboard_artifact.py:235]: Ahh got you. Thanks!

**Review by harupy — APPROVED** [review]
LGTM 🎉

**Pull request merged**